tests/core/test_settings.py

```diff
@@ -0,0 +1,186 @@
+
+            if save.upper() == "Y":
+                yaml = YAML()
+                with open(settings_path) as f:
+                    settings = yaml.load(f)
+                    assert settings["udemy"]["email"] == email
+                    assert settings["udemy"]["password"] == password
+                    assert settings["udemy"]["zipcode"] == zip_code
+                    assert (
+                        settings["udemy"]["languages"] == []
+                        if languages is None
+                        else ",".join(languages)
+                    )
+                    assert (
+                        settings["udemy"]["categories"] == []
+                        if categories is None
+                        else categories
+                    )
+                # Load settings just created
+                Settings(settings_path)
+            else:
+                assert os.path.isdir(settings_path) is False
+
+
+@pytest.mark.parametrize(
+    "email,password,zip_code,languages,categories,save,file_name",
+    [
+        (
+            "test9@mail.com",
+            "uherwh834",
+            "12345",
+            None,
+            None,
+            "Y",
+            "test_load_existing_settings1.yaml",
+        ),
+        (
+            "test80@mail.com",
+            "jkajsdsad",
+            "None",
+            "Italian",
+            None,
+            "Y",
+            "test_load_existing_settings9.yaml",
+        ),
+        (
+            "test10@mail.com",
+            "234sdfs",
+            "None",
+            "English",
+            "Development,Art",
+            "Y",
+            "test_load_existing_settings2.yaml",
+        ),
+        (
+            "test11@mail.com",
+            "frtuhrfty234",
+            "788192",
+            "French,German",
+            None,
+            "Y",
+            "test_load_existing_settings3.yaml",
+        ),
+    ],
+    ids=(
+        "load existing settings no languages",
+        "load existing settings no categories",
+        "load existing settings no zipcode",
+        "load existing settings full",
+    ),
+)
+def test_load_existing_settings(
+    email, password, zip_code, languages, categories, save, file_name
+):
+    with mock.patch(
+        "builtins.input", side_effect=[email, zip_code, languages, categories, save]
+    ):
+        with mock.patch("getpass.getpass", return_value=password):
+            settings_path = f"test_tmp/{file_name}"
+            Settings(settings_path)
+
+    # Load existing settings
+    settings = Settings(settings_path)
+    assert settings.email == email
+    assert settings.password == password
+    assert settings.zip_code == zip_code
+    assert settings.languages == [] if languages is None else languages
+    assert settings.categories == [] if categories is None else categories
+
+
+@pytest.mark.parametrize(
+    "is_ci_run,email,password",
+    [
+        (
+            True,
+            "username1@mail.com",
```

Comment:
are these real account ? can we somehow hide these ?

Reply:
No I don't think so. I just made them up for unittests

Reply:
oh ya the diff was too large and didn't load for me lol